rfc/0015-multi-file-jayvee/0015-multi-file-jayvee.md

```diff
@@ -0,0 +1,199 @@
+- elements within a pipeline can access
+  - root level elements by their name
+  - elements within the same pipeline by their name
+  - and elements of packages by their qualified name
+
+**No-access paths:**
+
+- elements within a pipeline cannot be referenced by outside elements
+
+**Used elements of different files are handled as if they were defined at the root level.**
+
+### Publishing elements for usage elsewhere (within the project)
+
+For publishing single elements, the RFC introduces the keyword `publish` to indicate the visibility `file-published`.
+All elements within a file are not published per default, visibility `file-private`.
+Explicitly declaring an element as published allows for usage elsewhere.
+
+**Example publish**
+
+```
+// define and publish in one syntax
+publish valuetype MyValueType1 {
+  // ... details
+}
+
+// define first
+valuetype MyValueType2 {
+  // ... details
+}
+
+// publish later
+publish MyValueType2;
+
+// publish later under a different name
+publish MyValueType2 as MyValueType3;
+```
+
+### Packages: bundling elements to a package for decoupled usage
+
+For bundling and publishing elements, the RFC introduces a new concept called `packages`.
+A `package` can inhibit `Valuetype`s, `Block`s, `BlockType`s, `Constraint`s, `Transform`s, and further `Package`s.
+A package must be of visibility `file-published` and, thus, requires the keyword `publish`.
+Elements within a package can use the `publish` keyword to make them of visibility `package-published` or are `package-private` per default.
+
+**Example package**
+
+```
+publish package MyDomainPackage {
+  // definition of a new valuetype as part of the package
+  publish valuetype MyDomainSpecificValuetype1 {
+    // ... details of valuetype
+  }
+
+  // reference to an existing valuetype to make it part of the package
+  publish MyDomainSpecificValuetype2;
+
+  // ... possibly more elements
+}
+```
+
+The advantage of bundling elements into a `package` is the decoupling from the internal file system structure.
+Rather than accessing files directly (and needing knowledge what is element is located in which files) users can simply use a whole package with all its elements (and don't need to know in which file the element is originally defined).
+
+### Using elements
+
+Only `file-published` (with keyword `publish`) elements can be used in other files.
+
+#### Usage paths
+
+When using elements of a file or a package, we have to define where the elements are located.
+Jayvee provides the following possibilities:
+
+- a relative file path, e.g., `use * from './path/to/file.jv';`
+
+The `use` of elements via a file path decouples by using the element name or a defined alias instead of the file path within the file.
+
+#### Using published elements of a file (within the same project)
+
+```
+use * from './path/to/location.jv'; // use all published elements from the file, access via qualified name as if it would be defined at the root level
```

Comment:
What happens if `./path/to/location.jv` exports something with a already used name?

Reply:
Allowing `use * as XYZ from './path/to/location.jv'` might be a fallback in this case.